### PR TITLE
Update scanner start/stop logic

### PR DESCRIPTION
**Pull request by optisamit**

### CODE REVIEW

File: ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java [hunk 1]
```diff
@@ -768,12 +686,12 @@ public void onConnected(@NonNull Connector connector) {
         );
 
         Dispatch.post(() -> {
+            connectionsInProgress--;
+            if (connectionsInProgress == 0) { // We went from 1 to 0 - maybe time to stop scanning
```
Comment (therock24) [inline ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java]: here you mean start scanning

File: ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java [hunk 2]
```diff
@@ -876,18 +794,18 @@ public void onConnectionFailure(@NonNull Connector connector, UlxError error) {
...
-
-            // If we cannot restart the adapter - let's continue scanning
-            startScanning();
+        if (delegate != null) {
+            delegate.onAdapterRestartRequest(this);
         }
+
+        Dispatch.post(() -> {
+            connectionsInProgress--;
+            if (connectionsInProgress == 0) { // We went from 1 to 0 - maybe time to stop scanning
```
Comment (therock24) [inline ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java]: start scanning

File: ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java [hunk 3]
```diff
@@ -632,85 +618,19 @@ private void queueConnection(BluetoothDevice bluetoothDevice) {
...
-
-        // Request the connector at the top of the queue to connect
-        Timber.i(
-                "ULX connector queue proceeding with connector %s",
-                connector.getIdentifier()
-        );
         connector.connect();
+        Dispatch.post(() -> {
+            connectionsInProgress++;
+            if (connectionsInProgress == 1) { // We went from 0 to 1 - maybe time to start scanning
```
Comment (therock24) [inline ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java]: stop scanning

File: ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java (hunk 1 above)
Comment (optisamit) [reply ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java:690]: @therock24 Yes, if we are currently requested to scan (via `requestAdapterToStartBrowsing()`)

Comment (optisamit) [reply ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleBrowser.java]: Ah, yes. Good catch!

**Pull request merged**